Casks/d/daedalus-mainnet.rb

```diff
@@ -11,8 +11,11 @@
   livecheck do
     url "https://update-cardano-mainnet.iohk.io/daedalus-latest-version.json"
     regex(%r{/daedalus[._-](\d+(?:\.\d+)+)[._-](\d+)[._-]mainnet[._-](\h+)[._-]x86_64[._-]darwin\.pkg}i)
```

Comment:
```suggestion
    regex(/daedalus[._-]v?(\d+(?:\.\d+)+)[._-](\d+)(?:[._-]mainnet)?[._-](\h+)/i)
```

While we're working on this, there may be something to be said for only making this regex as explicit as it needs to be (i.e., glossing over suffixes that could change or disappear in the future). We can get away with a bit more looseness with the updated `strategy` block, since we'll only be matching within the darwin URL (instead of the entire JSON text).